easylinks/src/main/java/com/google/step/servlets/ManageServlet.java

```diff
@@ -57,18 +57,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   }
 
   private static void addLink(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Check the input strings are valid
     String shortcut = ServletHelper.getParameterWithDefault(request, "shortcut", "");
     String url = ServletHelper.getParameterWithDefault(request, "url", "");
     if (shortcut.isEmpty() || shortcut.startsWith(PROVIDED_LINK) || url.isEmpty()) {
       response.getWriter().println("Invalid Input.");
       return;
     }
 
+    // Check if the shortcut already created in the DataStore
+    String email = ServletHelper.USERSERVICE.getCurrentUser().getEmail();
+    if (ServletHelper.fetchUrlWithDefault(shortcut, email, null) != null) {
```

Comment:
fetchUrlWithDefault takes String defaultValue, not sure if null can be auto converted to a string in java? Maybe use "" instead of null?

Reply:
I've tested it. It can prevent the repeating shortcut in the datastore. It doesn't matter to assign a null value to a String, because String is an object that can be null.

Reply:
Yes, we could assign a null value to String. It doesn't matter.